packages/relay/src/lib/eth.ts

```diff
@@ -2331,15 +2331,22 @@ export class EthImpl implements Eth {
           transactionIndex: nullableNumberTo0x(receiptResponse.transaction_index),
         });
       });
-
+      const functionSelectors = constants.HTS_CREATE_FUNCTIONS_SIGNATURE.map((signature) =>
+        Utils.calculateFunctionSelector(signature),
+      );
+      const isCreationViaSystemContract = functionSelectors.includes(
+        receiptResponse.function_parameters.substring(0, constants.FUNCTION_SELECTOR_CHAR_LENGTH),
+      );
+      const tokenAddress = receiptResponse.call_result.substring(90);
+      const contractAddress = isCreationViaSystemContract ? prepend0x(tokenAddress) : receiptResponse.address;
```

Comment:
I think from `isCreationViaSystemContract` to `contractAddress` should be in a separate method that be easily tested.
Also you should rearrange it so that if `isCreationViaSystemContract` is false we exit early and return `receiptResponse.address`

Reply:
Done : )